components/Votes/ActiveVotes.tsx

```diff
@@ -0,0 +1,396 @@
+import { formatVotesToCommit } from "helpers";
+import { config } from "helpers/config";
+import {
+  useCommitVotes,
+  useContractsContext,
+  useDelegationContext,
+  usePanelContext,
+  useRevealVotes,
+  useStakingContext,
+  useUserContext,
+  useVotesContext,
+  useVoteTimingContext,
+  useWalletContext,
+} from "hooks";
+import { useEffect, useState } from "react";
+import { SelectedVotesByKeyT, VoteT } from "types";
+import {
+  ButtonInnerWrapper,
+  ButtonOuterWrapper,
+  ButtonSpacer,
+  Divider,
+  InfoText,
+  PaginationWrapper,
+  RecommittingVotesMessage,
+  Title,
+  VotesTableWrapper,
+  WarningIcon,
+} from "./style";
+
+export function ActiveVotes() {
+  const { activeVotesList, getUserDependentIsFetching } = useVotesContext();
+  const { phase, roundId } = useVoteTimingContext();
+  const { address, hasSigningKey, correctChainConnected, signingKey } =
+    useUserContext();
+  const { signer, sign, isSigning, setCorrectChain, isSettingChain } =
+    useWalletContext();
+  const { votingWriter } = useContractsContext();
+  const { stakedBalance } = useStakingContext();
+  const { getDelegationStatus, getDelegatorAddress } = useDelegationContext();
+  const { openPanel } = usePanelContext();
+  const [{ connecting: isConnectingWallet }, connect] = useConnectWallet();
+  const { commitVotesMutation, isCommittingVotes } = useCommitVotes();
+  const { revealVotesMutation, isRevealingVotes } = useRevealVotes();
+  const [selectedVotes, setSelectedVotes] = useState<SelectedVotesByKeyT>({});
+  const [dirtyInputs, setDirtyInput] = useState<boolean[]>([]);
+  const [votesToShow, setVotesToShow] = useState(activeVotesList);
+
+  useEffect(() => {
+    if (activeVotesList.length <= defaultResultsPerPage) {
+      setVotesToShow(activeVotesList);
+    }
+  }, [activeVotesList]);
+
+  function isDirty(): boolean {
+    return dirtyInputs.some((x) => x);
+  }
+
+  const isDelegate = getDelegationStatus() === "delegate";
+  const isDelegator = getDelegationStatus() === "delegator";
+  const delegatorAddress = isDelegate ? getDelegatorAddress() : undefined;
+
+  const actionStatus = calculateActionStatus();
+  type ActionStatus = {
+    tooltip?: string;
+    label: string;
+    infoText?: { label: string; tooltip: string };
+    onClick: () => void;
+    disabled?: boolean;
+    hidden?: boolean;
+    canCommit: boolean;
+    canReveal: boolean;
+  };
+  function calculateActionStatus(): ActionStatus {
+    const actionConfig: ActionStatus = {
+      hidden: true,
+      tooltip: undefined,
+      label: "",
+      infoText: undefined,
+      onClick: () => undefined,
+      disabled: true,
+      canCommit: false,
+      canReveal: false,
+    };
+
+    const isCommit = phase === "commit";
+    const isReveal = phase === "reveal";
+    const hasStaked = stakedBalance?.gt(0) ?? false;
+    const hasSigner = !!signer;
+
+    const hasPreviouslyCommittedAll =
+      activeVotesList.filter((vote) => vote.decryptedVote).length ===
```

Comment:
Previously we were iterating over the `votesToShow` variable in this logic. This actually would have caused a bug if there were enough active votes to show the pagination. It would have ignored all the votes besides the current page.